internal/plugin/extract.go

```diff
@@ -0,0 +1,83 @@
+package plugin
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+const pluginsExtractsFolder = pluginsFolder + "/extracts"
+
+func init() {
+	_ = os.MkdirAll(pluginsExtractsFolder, 0755)
+}
+
+func extractPlugin(filePath string) (*string, error) {
```

Comment:
Is there a reason why this is a `*string` instead of just a `string`?

Reply:
I wanted to return nil, but realize now that's redundant with returning a non-nil error, so instead just return an empty string on error.